app/controllers/application_controller.rb

```diff
@@ -1,21 +1,31 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
+
   # TODO: Remove when devise is added
-  def user_logged_in?
-    !!current_user
+  def authenticate_user!
+    raise "Not logged in" unless user_logged_in?
   end
 
   def current_user
-    User.first
+    @__current_user__ ||= User.first # rubocop:disable Naming/MemoizedInstanceVariableName
```

Comment:
I want to ensure we don't accidentally use `@current_user` rather than `current_user` anywhere atm, which is why I'm doing this. All of this gets removed once I add Devise, but I'm defering that until later in October.